src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/split_fc.cpp

```diff
@@ -0,0 +1,213 @@
+        int split_num = sub_stream_num + 1;
+
+        auto split_parts = [](int len, int n) {
+            int average = len / n;
+            std::vector<int> parts(n, average);
+            parts.back() = len - average * (n - 1);
+            return parts;
+        };
+
+        // TODO: support transpose
+        if (ov::is_type<ov::op::v1::Transpose>(fc_weight_node)) {
+            return false;
+        }
+
+        // 1. If the model is INT4 format, split the INT4 pattern for the FuseFCAndWeightsDecompression.
+        // 2. If the model is NOT INT4 format, split the weight.
+        std::vector<ov::Output<ov::Node>> wgt_node_vec(split_num);
+        if (ov::is_type<ov::op::v1::Multiply>(fc_weight_node) || ov::is_type<ov::op::v1::Reshape>(fc_weight_node)) {
+            // INT4 model should consider two patterns, including with Reshape Node and without Reshape Node.
+            const auto reshape_node = ov::as_type_ptr<ov::op::v1::Reshape>(fc_weight_node);
+            const auto multiply_node = reshape_node ? reshape_node->get_input_node_shared_ptr(0) : fc_weight_node;
+            if (!ov::is_type<ov::op::v1::Multiply>(multiply_node)) {
+                return false;
+            }
+            auto multiply_pattern = multiply_node->get_input_node_shared_ptr(1);
+            if (!ov::is_type<ov::op::v0::Constant>(multiply_pattern)) {
+                return false;
+            }
+            auto subtract_node = multiply_node->get_input_node_shared_ptr(0);
+            if (!ov::is_type<ov::op::v1::Subtract>(subtract_node)) {
+                return false;
+            }
+            auto convert_node1 = subtract_node->get_input_node_shared_ptr(1);
+            if (!ov::is_type<ov::op::v0::Convert>(convert_node1)) {
+                return false;
+            }
+            auto convert_node1_const = ov::as_type_ptr<ov::op::v0::Constant>(convert_node1->get_input_node_shared_ptr(0));
+            if (!convert_node1_const) {
+                return false;
+            }
+            auto convert_node0 = subtract_node->get_input_node_shared_ptr(0);
+            if (!ov::is_type<ov::op::v0::Convert>(convert_node0)) {
+                return false;
+            }
+            auto wgt_item = convert_node0->get_input_node_shared_ptr(0);
+            auto cvt_prec = convert_node0->get_element_type();
+
+            auto split_dim_range = wgt_item->get_shape()[split_dim];
+            const auto& convert_node1_shape = convert_node1->get_shape();
+            bool need_to_split_convert = ov::shape_size(convert_node1_shape) > 1 &&
+                                         split_dim < convert_node1_shape.size() &&
+                                         convert_node1_shape[split_dim] == split_dim_range;
+            const auto& wgt_shape = wgt_item->get_shape();
+
+            // needn't to split fc when the dim is 0.
+            if (split_dim_range <= 1 || ov::shape_size(wgt_shape) < 6600000) {
+                return false;
+            }
```

Comment:
> 1. We can easily move this check to common part to avoid duplication on L172 (weights decompression ops don't affect weights output shape, so we can just use `wgt_item->get_shape()` in the checks).
> 2. Could you please explain what the magic number `6600000` is? And please also add explanatory comment

1. updated.
2. `660000` is a threshold that is tested and summarized by existed KPI model. That is because some LLM's FC is small and split operation is degression.

Reply:
Please add a explanatory comment in the code in follow-up PR about this trade-off value

Reply:
> Please add a explanatory comment in the code in follow-up PR about this trade-off value

sure. I will do it.